src/praisonai-train/praisonai_train/cli/commands/train.py

```
@@ -79,8 +79,12 @@ def train_llm(
     try:
         praison = PraisonAI()
         praison.main()
-    except SystemExit:
-        pass
+    except SystemExit as exc:
+        # Propagate a non-zero exit (e.g. missing training dependencies) so
+        # callers and CI gating on ``$?`` see the failure. A clean exit is
+        # swallowed so the command returns 0 as before.
+        if exc.code:
+            raise typer.Exit(exc.code if isinstance(exc.code, int) else 1)
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)